### PR TITLE
fix sort behavior for root, add useful error, docs and tests

### PR DESCRIPTION
**Pull request by gmbecker**
Addressses all parts of #816

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -62,7 +62,6 @@ Encoding: UTF-8
 Language: en-US
 LazyData: true
 Roxygen: list(markdown = TRUE)
-RoxygenNote: 7.2.3
```
Comment (Melkiades) [inline DESCRIPTION:65]: I think at the moment, we use 7.3.0

File: R/tt_sort.R [hunk 2]
```diff
@@ -156,7 +180,16 @@ sort_at_path <- function(tt,
   }
 
   ## XXX hacky fix this!!!
+  ## tt_at_path removes root even if actual root table isn't named root, we need to match that behavior
+  if(path[1] == "root") {
```
Comment (Melkiades) [inline R/tt_sort.R:184]: I just fixed a problem with `as_result_df` for cases with `rbind`ed tables, and seems that there is a `"rbind_root"` at the beginning of the path. I removed all initial "root" and "rbind_root" from there when doing as_result_df (also because it is not really informative), but I was wondering if we need to check for this also here or if we should fix it in the `rbind` function

Comment (gmbecker) [reply R/tt_sort.R:184]: I'm not so sure. As I said, I actually think this behavior is outdated and should be deprecated (when the first element, whether root or rbind_root) doesn't actually match the path (ie there is no "root"/"rbind_root" subtable at the root level, its named something else). 

For when it DOES match, it already doesn't matter whether its there or not; you don't have to name the root element when specifying paths (I'm not sure if this is true everywhere... someone should check!, but its true of sort_at_path and tt_at_path, and I think the latter is used by a lot of the user-facing pathing support).

Put simply, I think we should be moving away from supporting users putting in paths that are *actively wrong*, since we now after this PR will give them a really nice error message that ells them how to figure out what paths are valid.

We (ie you guys) could also write a check_path_valid function that could be called at the top of things with pathing support, I suppose, but I'm not sure its worth a full additional walk of the tree, tbh.

File: DESCRIPTION (hunk 1 above)
Comment (gmbecker) [reply DESCRIPTION:65]: Why was this changed @jcheng5, @pawelru @cicdguy ? Are new features actually being used? At the very least it should probably be updated again to 7.3.1 which looks to contain a number of bugfixes of various significance. 

Also, fyi, because 7.3.0 is now not the latest CRAN release and so is only available in source form, AND it requires c++11 features, its pretty non-trivial to install (it currently won't install for me, I'll keep trying but its a lot of wheel spinning thats not really useful to anyone).

Comment (gmbecker) [reply DESCRIPTION:65]: Yep, sorry about that other Joe. my mistake there.

I think we should do that update, it seems 7.3.1 is purely bugfixes on top of 7.3.0. 

Also, @cicdguy is there anywhere convenient that the manifest for the images is published?  If not, its probably a good idea since roxygen mismatches cause check failures  the iteration overhead both for the contributor and the nest team is non-trivial. I'm not exactly sure where the best place to put that would be, though, as it would apply across all NEST projects, right? (though I don't know how many contributed PRs the other projects are getting...)

**Review by Melkiades — COMMENTED** [review]
For me this is good to go. Thank you @gmbecker!

@shajoezhu when you think is best we can merge this PR

**Review by shajoezhu — APPROVED** [review]
Thanks Gabe! I think this is good. Let's do one more check in the integration test before we merging this.

**Review by edelarua — APPROVED** [review]
No breaking changes in downstream packages. This should be good to go!

**Comment by shajoezhu** [comment]
good to go. thank you all @gmbecker @edelarua @Melkiades
